LibreNMS/Util/EnvHelper.php

```diff
@@ -119,7 +120,16 @@ public static function init()
             if (! file_exists($env_file)) {
                 copy(base_path('.env.example'), $env_file);
 
-                $key = trim(exec(PHP_BINARY . ' ' . base_path('artisan') . ' key:generate --show'));
+                $key = null;
+                if (php_sapi_name() == 'cli') {
```

Comment:
Why the if/else?  Could we not just use Artisan::call() 100% of the time?

Reply:
It's called by a composer hook here https://github.com/librenms/librenms/blob/0f37f68bfdfa5a30fbe2509c7eafc945a4b6df37/LibreNMS/ComposerHelper.php#L39

But I didn't test. Figured it doesn't have laravel booted